src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-smart-list/osd-smart-list.component.ts

```diff
@@ -0,0 +1,111 @@
+import { Component, Input, OnChanges, OnInit, SimpleChanges } from '@angular/core';
+import { I18n } from '@ngx-translate/i18n-polyfill';
+import * as _ from 'lodash';
+import {
+  OsdService,
+  SmartAttribute,
+  SmartDataV1,
+  SmartError
+} from '../../../../shared/api/osd.service';
+import { CdTableColumn } from '../../../../shared/models/cd-table-column';
+
+@Component({
+  selector: 'cd-osd-smart-list',
+  templateUrl: './osd-smart-list.component.html',
+  styleUrls: ['./osd-smart-list.component.scss']
+})
+export class OsdSmartListComponent implements OnInit, OnChanges {
+  @Input()
+  osdId: number;
+  loading = false;
```

Comment:
Please add an empty line here to separate component input variables from the others.